File: itexpert-content-ui/src/app/modeles/ContentNode.ts
import {ContentUrl} from "./ContentUrl";
import {Value} from "./Value";
import {Rule} from "./Rule";
import {ContentFile} from "./ContentFile";
import {Translation} from "./Translation";
import { LockInfo } from "./LockInfo";

export class ContentNode {

  public id: string;
  public parentCode: string;
  public parentCodeOrigin: string;

  public code: string;
  public slug: string;

  public environmentCode: string;

  public type: string;

  public language: string;
  public title: string;
  public description: string;
  public iconUrl: string;
  public redirectUrl: string;
  public pictureUrl: string;
  public content: string;
  public creationDate: number;
  public modificationDate: number;
  public modifiedBy: string;
  public urls: ContentUrl[];

  public snapshot: boolean;


  public file: ContentFile;

  public tags: string[];
  public values: Value[];
  public roles: string[];
  public rules: Rule[];

  public datas: Value[];

  public published: ContentNode;
  public publicationDate: number;
  public version: string;
  public status: string;
  public favorite: boolean;
  public publicationStatus: string;

  public translations: Translation[];

  public userName: string;

  public lockInfo: LockInfo;
}

File: itexpert-content-lib/src/main/java/com/itexpert/content/lib/models/ContentNode.java
package com.itexpert.content.lib.models;

import com.itexpert.content.lib.enums.ContentTypeEnum;
import com.itexpert.content.lib.enums.StatusEnum;
import lombok.Data;
import org.springframework.data.annotation.Id;

import java.io.Serializable;
import java.util.List;
import java.util.UUID;

@Data
public class ContentNode implements Serializable, Cloneable {

    @Id
    private UUID id;

    private String parentCode;

    private String parentCodeOrigin;

    private String code;

    private String environmentCode;

    private String language;

    private ContentTypeEnum type;

    private String title;
    private String description;
    private String redirectUrl;
    private String iconUrl;
    private String pictureUrl;

    private String content;

    private List<ContentUrl> urls;

    private ContentFile file;

    private List<String> tags;
    private List<Value> values;
    private List<String> roles;
    private List<Rule> rules;

    private Long creationDate;
    private Long modificationDate;
    private UUID modifiedBy;

    private String version;
    private Long publicationDate;
    private StatusEnum status;

    private boolean favorite;

    private String publicationStatus;

    private List<Translation> translations;

    public ContentNode clone() throws CloneNotSupportedException {
        return (ContentNode) super.clone();
    }
}
